ppb/systems.py

```diff
@@ -40,7 +57,61 @@ def on_update(self, update, signal):
         for pygame_event in pygame.event.get():
             ppb_event = self.event_map.get(pygame_event.type)
             if ppb_event is not None:
-                signal(ppb_event())
+                signal(getattr(self, ppb_event)(pygame_event, update.scene))
+
+    def quit(self, event, scene):
+        return events.Quit()
+
+    def mouse_motion(self, event, scene):
+        screen_position = Vector(*event.pos)
+        camera = list(scene.get(tag="main_camera"))[0]
+        game_position = camera.translate_to_frame(screen_position)
+        delta = Vector(*event.rel) * (1/camera.pixel_ratio)
+        buttons = [bool(x) for x in event.buttons]
+        return events.MouseMotion(game_position, screen_position, delta, buttons)
+
+class MouseSystem(System):
+    """
+    Mouse system.
+    """
+
+    def __init__(self, *, engine, resolution=default_resolution, **kwargs):
+        self.mouse = Mouse()
+        self.offset = Vector(-0.5 * resolution[0],
+                             -0.5 * resolution[1])
+        engine.register(events.Update, "mouse", self.mouse)
+
+    def activate(self, engine):
+        """Sync mouse with hardware."""
+        buttons = self.get_hardware_buttons()
```

Comment:
Does this even return a different length for `buttons` and, if so, what would happen?

Reply:
This is a fun one: It's going to be hardware lib (and probably actual hardware) dependent.

In a given run time? No, it shouldn't.

Between runtimes: Possibly.

That said, it should just work since it will be an array like object and the mouse can handle that.